tests/integration/runk/runk-tests.bats

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bats
+#
+# Copyright (c) 2023,2024 Kata Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# This test will validate runk with containerd
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/../../metrics/lib/common.bash"
+
+setup_file() {
+	export RUNK_BIN_PATH="/usr/local/bin/runk"
+	export TEST_IMAGE="docker.io/library/busybox:latest"
+	export CONTAINER_ID="id1"
+	export PID_FILE="${CONTAINER_ID}.pid"
+	export WORK_DIR="${BATS_FILE_TMPDIR}"
+
+	echo "pull container image"
+	check_images ${TEST_IMAGE}
+}
+
+setup() {
+	# Bind mount ${WORK_DIR}:/tmp. Tests below will store files in this dir and check them when container is frozon.
+	sudo ctr run --pid-file ${PID_FILE} -d \
+		--mount type=bind,src=${WORK_DIR},dst=/tmp,options=rbind:rw \
+		--runc-binary ${RUNK_BIN_PATH} \
+		${TEST_IMAGE} \
+		${CONTAINER_ID}
+	read CID PID STATUS <<< $(sudo ctr t ls | grep ${CONTAINER_ID})
+	# Check the pid is consistent
+	[ "${PID}" == "$(cat "${PID_FILE}")" ]
+	# Check the container status is RUNNING
+	[ "${STATUS}" == "RUNNING" ]
+}
+
+teardown() {
```

Comment:
mmm why not to use https://github.com/kata-containers/kata-containers/blob/main/tests/common.bash#L182?

Reply:
hi @GabyCT , you mean use `check_processes` to ensure that shimv2 process really got killed?